clients/src/main/java/org/apache/kafka/clients/admin/Admin.java

```diff
@@ -1660,6 +1663,34 @@ default FenceProducersResult fenceProducers(Collection<String> transactionalIds)
     FenceProducersResult fenceProducers(Collection<String> transactionalIds,
                                         FenceProducersOptions options);
 
+    /**
+     * Determines the client's unique client instance ID used for telemetry. This ID is unique to
+     * this specific client instance and will not change after it is initially generated.
+     * The ID is useful for correlating client operations with telemetry sent to the broker and
+     * to its eventual monitoring destinations.
+     *
+     * If telemetry is enabled, this will first require a connection to the cluster to generate
+     * the unique client instance ID. This method waits up to {@code timeout} for the admin
+     * client to complete the request.
+     *
+     * If telemetry is disabled, the method will throw {@link IllegalStateException}.
```

Comment:
Do we need to add this? It's already explained below via `@throws...`

Reply:
I thought about it and kept it continue the disabled case from previous para. But it might be good to remove it as it's redundant.

Removed, done at all places.